### ISSUE TITLE
Potential MT optimizations

### ISSUE THREAD

**Issue by richardhendricks** [GTNewHorizons/NewHorizons#5175]
After working on the Forestry worktable speedup, I have an idea on how we can improve MT script processing time.

Looking at the code for Minetweaker, it looks like it reads an item from the script, does the change, reads the next item, etc.

However, this is inefficient since it puts the new item at the end of the recipelist and re-scans it again.

So my proposal is this - have it skip over items it has already completed. This will need to be done differently for replacements and outright recipe deletions.

For deletions:
```
Create a list of the recipes to delete.
For each item in the recipeList
    Compare against the deleteList
    If it is supposed to be deleted, add the index to the front of a list of indexToDelete
For each item in the indexToDeleteList
  Delete that index from the recipeList // This is done back-to-front so that indexes stay valid in the array
```

For replacements:
```
start = 0
For each MT item (assuming replacement, use index access not iterator)
  for start to recipeList.size() (use index accesses, not iterator)
    When found
      remove old recipe from the recipeList
      add new recipe to the front
      If there are no wildcards, break out of inner loop
```

Add a new "feature" - move to front
Basically, given a list of recipes, move them to the front.
Same as "for replacements" but the old and new recipes are the same.

Would allow us to place frequently used recipes at the start of the recipemap. Stuff that people would shift-click or make in the crafting table in volume. Like polyethylene dust to tiny dusts for TNT, woven cloth, low tier pipes like wooden/bronze/iron/steel, maybe some block coloring ones, etc.

**Comment by Dream-Master** [comment]
@richardhendricks bart put 80% recipes in Code. This will speed up loading alot

**Comment by richardhendricks** [comment]
From what I understand, he is having problems with that?  Anyways, this change will help with MT performance.
